opencart 2.3.0.0 - 2.3.0.2/upload/catalog/controller/extension/smailyforopencart/cron_cart.php

```diff
@@ -15,71 +15,146 @@ public function index() {
-            // Get abandoned carts.
-            $abandoned_carts = $this->model_extension_smailyforopencart_helper->getAbandonedCarts();
+        if (!array_key_exists('smaily_for_opencart_enable_abandoned', $settings) ||
+            (int) $settings['smaily_for_opencart_enable_abandoned'] !== 1
+        ) {
+            echo('Abandoned cart disabled!');
+            die(1);
+        }
 
-            if (!empty($abandoned_carts)) {
-                foreach ($abandoned_carts as $cart) {
-                    // Addresses array for smaily api call.
-                    $addresses = array(
-                        'email' => $cart['email'],
-                        'firstname' => isset($cart['firstname']) ? $cart['firstname'] : '',
-                        'lastname' => isset($cart['lastname']) ? $cart['lastname'] : '',
-                    );
-                    // Sync values selected from admin panel.
-                    $cart_sync_values = $this->model_extension_smailyforopencart_helper->getAbandonedSyncFields();
-                    // Populate products list with empty values for legacy api.
-                    foreach ($cart_sync_values as $sync_value) {
-                        for ($i=1; $i < 11; $i++) {
-                            $addresses['product_' . $sync_value . '_' . $i] = '';
-                        }
-                    }
+        // Get abandoned carts.
+        $abandoned_carts = $this->model_extension_smailyforopencart_helper->getAbandonedCarts();
+        if (empty($abandoned_carts)) {
+            echo('No abandoned carts');
+            die();
+        }
 
-                    // Populate addresses fields with up to 10 products.
-                    $j = 1;
-                    foreach ($cart['products'] as $product) {
-                        if ($j <= 10) {
-                            foreach ($cart_sync_values as $sync_value) {
-                                if ($sync_value === 'quantity') {
-                                    $addresses['product_' . $sync_value . '_' . $j] = $product[$sync_value];
-                                } else {
-                                    $addresses['product_' . $sync_value . '_' . $j] = $product['data'][$sync_value];
-                                }
-                            }
-                            $j++;
-                        }
-                    }
+        // Sync values selected from admin panel.
+        $cart_sync_values = $this->model_extension_smailyforopencart_helper->getAbandonedSyncFields();
+        $fields_available = [
+            'name',
+            'description',
+            'sku',
+            'quantity',
+            'price',
+            'base_price'
+        ];
+        $selected_fields = array_intersect($fields_available, $cart_sync_values);
+
+        // Send out cart emails.
+        foreach ($abandoned_carts as $cart) {
+            // Addresses array for smaily api call.
+            $addresses = array(
```

Comment:
I think it would be better for clarity to rename this variable to `$address`. With `$addresses` one could assume that it contains data for multiple email addresses, but in reality it contains only for one.

Reply:
Changed in https://github.com/sendsmaily/smaily-opencart-module/pull/31/commits/5fa56965c805898fd437bd4b05d2466acdfd40f7